### ISSUE TITLE
[BUG] Upgrading some packages via pip fails with error message related to setuptools 60.9.3. Downgrading setuptools to 60.8.2 works.

### ISSUE THREAD

**Issue by SteffenBauer** [pypa/setuptools#3173]
### setuptools version

60.9.3

### Python version

Python 3.7.3

### OS

Debian 10

### Description

Since upgrading `setuptools` to the newest version, upgrading certain packages via `pip` fail. See below for console output.

Affected packages in particular are `srt` and `neo4j`. Both upgrades fail with same error `error: invalid command 'egg_info'`, when trying to build the `.whl`; see below console output.

Console output strongly hints that the issue might be somewhere in `setuptools`. Also downgrading `setuptools` to 60.8.2 solves the problem; this also strongly hints that something was broken in the newest version `setuptools` 60.9

There are already several issues reported here regarding `egg`; unfortunately I'm not too much of an expert here and cannot estimate if my problem is connected to these issues. I hope that this bug report is welcome anyway and provides useful information.

### Expected behavior

Upgrading via pip should work as it does with lower versions of setuptools, without throwing `error: invalid command 'egg_info'`

### How to Reproduce

1. pip install setuptools==60.9.3
2. pip install -U srt==3.5.2 neo4j==4.4.2
3. [upgrade fails with several warnings and error]
4. pip install setuptools==60.8.2
5. pip install -U srt==3.5.2 neo4j==4.4.2
6. [upgrade now works]

### Output

Output for srt package:
```console
Collecting srt
  Downloading srt-3.5.2.tar.gz (24 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      /home/steffen/.local/lib/python3.7/site-packages/setuptools/_distutils/dist.py:275: UserWarning: Unknown distribution option: 'python_requires'
        warnings.warn(msg)
      usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
         or: setup.py --help [cmd1 cmd2 ...]
         or: setup.py --help-commands
         or: setup.py cmd --help
      
      error: invalid command 'egg_info'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

Output for neo4j package:
```console
Collecting neo4j
  Using cached neo4j-4.4.2.tar.gz (89 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [12 lines of output]
      /home/steffen/.local/lib/python3.7/site-packages/setuptools/_distutils/dist.py:275: UserWarning: Unknown distribution option: 'install_requires'
        warnings.warn(msg)
      /home/steffen/.local/lib/python3.7/site-packages/setuptools/_distutils/dist.py:275: UserWarning: Unknown distribution option: 'entry_points'
        warnings.warn(msg)
      /home/steffen/.local/lib/python3.7/site-packages/setuptools/_distutils/dist.py:275: UserWarning: Unknown distribution option: 'python_requires'
        warnings.warn(msg)
      usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
         or: setup.py --help [cmd1 cmd2 ...]
         or: setup.py --help-commands
         or: setup.py cmd --help
      
      error: invalid command 'egg_info'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by SteffenBauer** [comment]
Thanks a lot for looking into it. I investigated much in the last hour, and also couldn't reproduce it with a docker environment.
I then tried to uninstall and reinstall the `setuptools` to get a complete fresh start, but pip then failed reinstalling `setuptools`.
I found an artifact `setuptools-57.4.0.dist-info` in my `.local/lib/python3.7/site-packages`. It seems that somehow something broke some time ago during update of my python system.
After removing this artifact directory then I could successfully reinstall `setuptools`, and then an installation of `srt` and `neo4j` worked without problem.

So issue resolved, reason was probably this corruption in my python lib.
